### PR TITLE
website: Add delete message feature for admins

### PR DESCRIPTION
**Pull request by rjmacarthy**
This PR addresses point number two for issue #1022 which add the ability to delete a message from the actions drop-down menu.  The back-end requires some work for individual message not to return after being deleted.

Any questions or concerns please let me know.

### CODE REVIEW

File: website/src/pages/api/messages/[id]/index.ts [hunk 1]
```diff
@@ -2,12 +2,48 @@ import { withoutRole } from "src/lib/auth";
...
+    const client = createApiClientFromUser(user);
+    await client.delete_message(id);
+    res.status(200).json({ message: "Message deleted" });
+  } catch (e) {
+    res.status(500).json({ message: "Failed to delete message" });
+  }
+};
+
+const handlers = {
+  GET: getMessage,
```
Comment (fozziethebeat) [inline website/src/pages/api/messages/[id]/index.ts:36]: I feel like this is kinda confusing.  I would much rather we have a pure delete API route in `website/src/pages/api/admin/delete_message.ts`.

That way you don't need all this meta handler wrappers and the handler can directly use `withRole('admin')`.

**Review by fozziethebeat — CHANGES_REQUESTED** [review]

File: backend/oasst_backend/prompt_repository.py [hunk 2]
```diff
@@ -673,7 +673,8 @@ def fetch_message(self, message_id: UUID, fail_if_missing: bool = True) -> Optio
         messages = self._add_user_emojis_all(qry)
         message = messages[0] if messages else None
 
-        message = self.db.query(Message).filter(Message.id == message_id).one_or_none()
+        message = self.db.query(Message).filter(Message.id == message_id, not_(Message.deleted)).one_or_none()
```
Comment (fozziethebeat) [inline backend/oasst_backend/prompt_repository.py:676]: Oof, i can't approve for this.  You'll need to get the backend team to review this (or make it a separate PR)

**Review by fozziethebeat — APPROVED** [review]

File: backend/oasst_backend/prompt_repository.py (hunk 2 above)
Comment (rjmacarthy) [reply backend/oasst_backend/prompt_repository.py:676]: Understood. @andreaskoepf, please could you take a look?

Comment (andreaskoepf) [reply backend/oasst_backend/prompt_repository.py:676]: FetchMessage is a low level function that needs to be able to pull every message out of the db. I will make changes for deleted messages later if that should be necessary. Please restore this file for now.

**Pull request merged**